public-docs/swag-shop-03-custom-styling.md

```diff
@@ -0,0 +1,61 @@
+---
+id: swag-shop-3
+title: Custom Styling
+---
+
+# Adding custom styling
+
+## Styling a existing component
+All the CSS styles can be added directly in `lib/reactionTheme.js` file.
+For example:
+To style the Header component the steps are:
+1. Open the Header component at `/src/components/Header/Header.js` and take note of the `name` of the Component passes to `withStyle`. Here it is "Header" as is evident from the line `@withStyles(styles, { name: "Header" })`
+2. Open `lib/reactionTheme.js` and in the object being passed to `createMuiTheme` add a property `overrides` in the following way:
+```javascript
+const theme = createMuiTheme({
+  overrides: {
+    Header: {
+      title: {
+        color: "black",
+        textTransform: "uppercase",
+        borderBottom: "2px"
+      }
+    },
+    layout: {.....},
+    .....
+  }
+```
+3. Refresh the webpage for the changes to load.
+
+## Styling a new component
+Either you can follow the above approach of naming the component and writing the styles in the `lib/reactionTheme.js` file.
+Or you can write the styling code within the component. For example we wrote the folloing component(HomePageStaticText) to display static text on the homepage:
+1. In the component file define a style object like:
+```javascript
+const styles = (theme) => ({
+  root: {
+    textAlign: "center",
+    marginBottom: 70
+  },
+  heading: {
+    fontSize: 25
+  },
+  text: {
+    fontSize: 28
+  }
+});
+```
+1. Add a @withStyle notation to the component class and pass the style object to it like: `@withStyles(styles, { name: "HomePageStaticText" })`
+2. Now assign classes to the sub-components in the `render()` like:
+```javascript
+render() {
+    const { classes } = this.props;
+    return (
+      <div className={classes.root}>
+        <h1 className={classes.heading}>#MADEINTHESUNSHINE</h1>
+        <p className={classes.text}>There is no one who loves pain itself, wants to have it, simply because it is pain...</p>
+        <p className={classes.text}>There is no one who loves pain itself</p>
+      </div>
+    );
+}
+```
```

Comment:
We can also steal the `Changing theme variables` section from the starterkit docs to explain how to change the Material-UI variables: https://github.com/reactioncommerce/reaction-next-starterkit/blob/develop/docs/theming.md

Reply:
cc @mikemurray

Reply:
I don't know that those docs are correct any longer

Reply:
These docs are up to date! Mike just updated them 5 days ago. But some of the parts might be different from the Swag Shop repo if it's not using Material UI or Material UI theme variables.

Reply:
Nevermind, I see what you are saying now